### PR TITLE
Enable safe_browsing_api for muon

### PR DESCRIPTION
**Pull request by jumde**

### CODE REVIEW

File: atom/browser/browser.cc [hunk 1]
```diff
@@ -156,6 +156,8 @@ void Browser::WillFinishLaunching() {
 
 void Browser::DidFinishLaunching(const base::DictionaryValue& launch_info) {
   is_ready_ = true;
+  sb_service = safe_browsing::SafeBrowsingService::CreateSafeBrowsingService();
```
Comment (bridiver) [inline atom/browser/browser.cc]: I think this should go in chromium_src/chrome/browser/browser_process_impl.cc to match the implementation in src/chrome/browser/browser_process_impl.cc

File: chromium_src/BUILD.gn [hunk 2]
```diff
@@ -290,6 +295,113 @@ source_set("browser") {
     "chrome/browser/content_settings/host_content_settings_map_factory.cc",
     "chrome/browser/content_settings/host_content_settings_map_factory.h",
 
+    "//chrome/browser/renderer_preferences_util.cc",
```
Comment (bridiver) [inline chromium_src/BUILD.gn]: can we add `//chrome/browser/safe_browsing` as a dep instead?

File: chromium_src/chrome/browser/safe_browsing/incident_reporting/binary_integrity_analyzer.cc [hunk 3]
```diff
@@ -0,0 +1,60 @@
+// Copyright 2014 The Chromium Authors. All rights reserved.
```
Comment (bridiver) [inline chromium_src/chrome/browser/safe_browsing/incident_reporting/binary_integrity_analyzer.cc]: why are these copied instead of using chrome src? Are there differences?

File: atom/browser/browser.cc (hunk 1 above)
Comment (jumde) [reply atom/browser/browser.cc]: I am working on the changes, I realized this yesterday. This is still in progress. Should've added the WIP tag to the header.

**Comment by bridiver** [comment]
I added the wip label, so just remove that when you're ready to review again

File: atom/browser/BUILD.gn [hunk 4]
```diff
@@ -21,22 +21,29 @@ source_set("browser") {
...
     "//components/prefs",
     "//components/metrics",
+    "//components/safe_browsing/common:safe_browsing_prefs",
     ":importer",
     "//electron/vendor/ad-block/muon:ad_block",
     "//electron/vendor/tracking-protection/muon:tp_node_addon",
   ]
 
   sources = [
+    "//chrome/browser/loader/safe_browsing_resource_throttle.h",
```
Comment (bridiver) [inline atom/browser/BUILD.gn]: these should go in chromium_src/BUILD.gn prerender target

File: atom/browser/BUILD.gn [hunk 5]
```diff
@@ -21,22 +21,29 @@ source_set("browser") {
   ]
 
   deps = [
+    "../../chromium_src/chrome/browser/prerender",
     "//electron:common",
     "//electron/muon/app",
     # @todo(bridiver) fix circular dep
     # "//electron/muon/browser",
     "//base",
+    "//chrome/browser:resource_prefetch_predictor_proto",
```
Comment (bridiver) [inline atom/browser/BUILD.gn]: these should be deps of the the chromium_src prerender target

File: atom/browser/BUILD.gn [hunk 6]
```diff
@@ -21,22 +21,29 @@ source_set("browser") {
   ]
 
   deps = [
+    "../../chromium_src/chrome/browser/prerender",
```
Comment (bridiver) [inline atom/browser/BUILD.gn]: deps should always absolute paths `//electron/chromium_src...`

File: chromium_src/BUILD.gn [hunk 7]
```diff
@@ -290,8 +290,118 @@ source_set("browser") {
     "chrome/browser/content_settings/host_content_settings_map_factory.cc",
     "chrome/browser/content_settings/host_content_settings_map_factory.h",
 
-    "//chrome/browser/ssl/security_state_tab_helper.cc",
+    "//chrome/browser/renderer_preferences_util.cc",
+    "//chrome/browser/renderer_preferences_util.h",
+
+    "//chrome/browser/safe_browsing/chunk_range.cc",
```
Comment (bridiver) [inline chromium_src/BUILD.gn]: I would put these in a separate safe_browsing target and make that a dep of `browser`

File: atom/browser/atom_download_manager_delegate.cc [hunk 8]
```diff
@@ -11,23 +11,294 @@
...
+  SafeBrowsingState() {}
+  ~SafeBrowsingState() override;
+
+ private:
+  DISALLOW_COPY_AND_ASSIGN(SafeBrowsingState);
+};
+
+SafeBrowsingState::~SafeBrowsingState() {}
+
+void CheckDownloadUrlDone(
```
Comment (bridiver) [inline atom/browser/atom_download_manager_delegate.cc]: seems like we're copying a lot of code from src/chrome here which we really try to avoid. Can we subclass or use some other method to reuse the existing code?

File: patches/master_patch.patch [hunk 9]
```diff
@@ -159,6 +159,159 @@ index 751043efd8038573b2870dc60e5d7ad737887d5f..321971b975659028005aabd8a9163ac8
...
++    : next_state_(STATE_CHECK_DOWNLOAD_URL),
+       confirmation_reason_(DownloadConfirmationReason::NONE),
+       should_notify_extensions_(false),
+       create_target_directory_(false),
+@@ -278,15 +277,7 @@ DownloadTargetDeterminer::Result
+   DCHECK(!virtual_path_.empty());
+ 
+   next_state_ = STATE_RESERVE_VIRTUAL_PATH;
+-
+-  if (!should_notify_extensions_ ||
```
Comment (bridiver) [inline patches/master_patch.patch]: can you provide some explanation for these? This is adding a lot of patches and it doesn't seem like they should all be necessary

File: muon/browser/muon_browser_process_impl.cc [hunk 10]
```diff
@@ -58,9 +93,20 @@ MuonBrowserProcessImpl::component_updater() {
   return component_updater(component_updater_, false);
 }
 
+void MuonBrowserProcessImpl::ApplyAllowCrossOriginAuthPromptPolicy() {
```
Comment (bridiver) [inline muon/browser/muon_browser_process_impl.cc]: is this part of safe browsing?

File: chromium_src/BUILD.gn [hunk 11]
```diff
@@ -599,6 +612,144 @@ source_set("browser") {
   }
 }
 
+source_set("safe_browsing") {
+  sources = [
+    "//chrome/browser/download/download_completion_blocker.cc",
```
Comment (bridiver) [inline chromium_src/BUILD.gn]: safe browsing has an existing target in chrome so I don't think we should have to add all these files individually

File: atom/browser/extensions/atom_component_extensions.h [hunk 12]
```diff
@@ -1,13 +1,19 @@
+// Copyright 2016 Brave authors. All rights reserved.
```
Comment (jumde) [inline atom/browser/extensions/atom_component_extensions.h]: This change was to fix the issues in the `pre-commit checks` . I can create a separate PR for the fixes if needed.

File: atom/browser/api/atom_api_download_item.cc [hunk 13]
```diff
@@ -78,9 +79,10 @@ DownloadItem::~DownloadItem() {
 }
 
 void DownloadItem::OnDownloadUpdated(content::DownloadItem* item) {
-  if (download_item_->IsDone()) {
-    Emit("done", item->GetState());
-
+  if (download_item_->IsDangerous()) {
+    Emit("done", 4, item->IsDangerous());
```
Comment (darkdh) [inline atom/browser/api/atom_api_download_item.cc]: why hardcoded `DownloadState::MAX_DOWNLOAD_STATE`?

File: atom/browser/atom_download_manager_delegate.cc [hunk 14]
```diff
@@ -92,7 +384,7 @@ void AtomDownloadManagerDelegate::OnDownloadPathGenerated(
   file_dialog::DialogSettings settings;
   settings.parent_window = window;
   settings.title = item->GetURL().spec();
-  settings.default_path = target_path;
+  settings.default_path = path;
   if (path.empty() && file_dialog::ShowSaveDialog(settings, &path)) {
```
Comment (darkdh) [inline atom/browser/atom_download_manager_delegate.cc]: @bridiver we should merged https://github.com/brave/muon/pull/385 first and then rebase this PR on it

File: patches/master_patch.patch [hunk 15]
```diff
@@ -201,6 +281,105 @@ index 247d24220beb7d176b890d7c2d45bdd2dfd8db99..c270bba0fabb557e8d3d1df8ac166bbb
...
+-#endif
+-
+ #if BUILDFLAG(ENABLE_CAPTIVE_PORTAL_DETECTION)
+ #include "chrome/browser/ssl/captive_portal_metrics_recorder.h"
+ #endif
+@@ -59,58 +55,8 @@ void ChromeMetricsHelper::RecordExtraShutdownMetrics() {
+ 
+ void ChromeMetricsHelper::RecordExtraUserDecisionMetrics(
+     security_interstitials::MetricsHelper::Decision decision) {
+-#if BUILDFLAG(ENABLE_EXTENSIONS)
```
Comment (darkdh) [inline patches/master_patch.patch]: we can use `#if BUILDFLAG(ENABLE_EXTENSIONS) && defined(MUON_CHROMIUM_BUILD)` to reduce the patch size

File: patches/master_patch.patch [hunk 16]
```diff
@@ -509,6 +688,80 @@ index 7ee1ec91e6ecb8cd9e237f0e67063ed704566ea6..9198607c50f23d7757016dcfb7526993
...
+ void DownloadUrlSBClient::IdentifyReferrerChain() {
+diff --git a/chrome/browser/safe_browsing/safe_browsing_blocking_page.cc b/chrome/browser/safe_browsing/safe_browsing_blocking_page.cc
+index 5b50863736f2306202e8345d83be2f399ed2c371..655cb90ef69156802b1f227f7f31210c6d83beae 100644
+--- a/chrome/browser/safe_browsing/safe_browsing_blocking_page.cc
++++ b/chrome/browser/safe_browsing/safe_browsing_blocking_page.cc
+@@ -194,20 +194,6 @@ void SafeBrowsingBlockingPage::FinishThreatDetails(const base::TimeDelta& delay,
+   // shutting down.
+   if (!g_browser_process->safe_browsing_service())
+     return;
+-
```
Comment (darkdh) [inline patches/master_patch.patch]: ditto for block commenting
```c
#if defined(MUON_CHROMIUM_BUILD)
#endif
```

File: atom/browser/atom_download_manager_delegate.cc (hunk 8 above)
Comment (jumde) [reply atom/browser/atom_download_manager_delegate.cc]: Most of the code is similar to `chrome_download_manager_delegate.cc`. I optimized some of it, let me know if you think we can optimize more.

File: atom/browser/api/atom_api_download_item.cc [hunk 17]
```diff
@@ -78,9 +79,10 @@ DownloadItem::~DownloadItem() {
 }
 
 void DownloadItem::OnDownloadUpdated(content::DownloadItem* item) {
-  if (download_item_->IsDone()) {
-    Emit("done", item->GetState());
-
+  if (download_item_->IsDangerous()) {
```
Comment (bridiver) [inline atom/browser/api/atom_api_download_item.cc]: will `IsDone` never return true if the download is dangerous?

File: patches/master_patch.patch [hunk 18]
```diff
@@ -1801,6 +1980,30 @@ index c921e6f316ee9a00c6b48270f84167c22deb8ce6..36d51e77b2ca10d45fe44636b9820f74
...
    return switches::autoplay::kNoUserGestureRequiredPolicy;
+diff --git a/net/base/linked_hash_map.h b/net/base/linked_hash_map.h
+index 921764f79c603fd8b8037ac52ca2ebabb0440ebd..a7f71c080f52e35310d90b6815b1c1a190ed1063 100644
+--- a/net/base/linked_hash_map.h
++++ b/net/base/linked_hash_map.h
+@@ -142,8 +142,7 @@ class linked_hash_map {
+   // map and list, a CHECK() error will occur.
+   iterator erase(iterator position) {
+     typename MapType::iterator found = map_.find(position->first);
+-    CHECK(found->second == position)
```
Comment (bridiver) [inline patches/master_patch.patch]: this seems unrelated?

File: patches/master_patch.patch [hunk 19]
```diff
@@ -512,6 +571,113 @@ index d85ddf9db90d332a31f839e7600b8d6e5ec61356..2dbe745b19001ce55631e625b6c597b2
...
    int FindTabStripModelById(int32_t tab_id, TabStripModel** model) const;
+diff --git a/chrome/browser/safe_browsing/chrome_password_protection_service.cc b/chrome/browser/safe_browsing/chrome_password_protection_service.cc
+index 4275342f9244aa005521882c23767662fa2a681c..fc68b078ae1438a66e0bd7fe76b847cae8db9f2c 100644
+--- a/chrome/browser/safe_browsing/chrome_password_protection_service.cc
++++ b/chrome/browser/safe_browsing/chrome_password_protection_service.cc
+@@ -239,13 +239,6 @@ void ChromePasswordProtectionService::ShowModalWarning(
+     web_contents->ExitFullscreen(true);
+ 
+   UpdateSecurityState(SB_THREAT_TYPE_PASSWORD_REUSE, web_contents);
+-  ShowPasswordReuseModalWarningDialog(
```
Comment (bridiver) [inline patches/master_patch.patch]: we need to reduce the size of this patch. Can you provide some more info about the reasons for the patches?

File: patches/master_patch.patch [hunk 20]
```diff
@@ -162,6 +162,65 @@ index e2faffbfcb27da0329d3e388fea8758d49139e8e..d8cdceffbc4b048395957dcae935a4c4
...
          ],
+diff --git a/chrome/browser/download/download_target_determiner.cc b/chrome/browser/download/download_target_determiner.cc
+index 4f4dadd98fd997e9659d10e6989916dbb0bb899e..72a3f10900c19a6e4cc544bbec52a4a3a451af59 100644
+--- a/chrome/browser/download/download_target_determiner.cc
++++ b/chrome/browser/download/download_target_determiner.cc
+@@ -92,7 +92,7 @@ DownloadTargetDeterminer::DownloadTargetDeterminer(
+     DownloadPrefs* download_prefs,
+     DownloadTargetDeterminerDelegate* delegate,
+     const CompletionCallback& callback)
+-    : next_state_(STATE_GENERATE_TARGET_PATH),
```
Comment (bridiver) [inline patches/master_patch.patch]: same as below, we really want to avoid patching like this

File: muon/browser/muon_browser_process_impl.cc [hunk 21]
```diff
@@ -6,27 +6,56 @@
...
   device_client_.reset(new ChromeDeviceClient);
 }
 
 MuonBrowserProcessImpl::~MuonBrowserProcessImpl() {
+  if (safe_browsing_service_.get())
+    safe_browsing_service()->ShutDown();
   g_browser_process = NULL;
 }
 
+safe_browsing::SafeBrowsingService*
```
Comment (bridiver) [inline muon/browser/muon_browser_process_impl.cc]: these should go in chromium_src browser_process_impl
muon_browser_process_impl is for muon extensions that aren't part of chrome browser_process_impl

File: atom/browser/native_window.cc [hunk 22]
```diff
@@ -94,7 +94,7 @@ NativeWindow::NativeWindow(
   // add to browser list
   ::Browser::CreateParams create_params(::Browser::Type::TYPE_TABBED,
       Profile::FromBrowserContext(inspectable_web_contents_->
-          GetWebContents()->GetBrowserContext()));
+          GetWebContents()->GetBrowserContext()), false);
```
Comment (bridiver) [inline atom/browser/native_window.cc]: always setting this to false seems problematic because it ignores the actual user gesture status. Where is this used?

File: atom/browser/atom_download_manager_delegate.cc [hunk 23]
```diff
@@ -136,32 +320,24 @@ void AtomDownloadManagerDelegate::OnDownloadPathGenerated(
...
       ui::SelectFileDialog::SELECT_SAVEAS_FILE,
       base::Bind(&AtomDownloadManagerDelegate::OnDownloadItemSelected,
                  base::Unretained(this), callback, download_item),
       base::Bind(&AtomDownloadManagerDelegate::OnDownloadItemSelectionCancelled,
                  base::Unretained(this), callback, item));
-  } else {
-    if (download_item)
-      download_item->SetSavePath(path);
-
-    callback.Run(path,
```
[the remark this answers is not in the document]
Comment (darkdh) [reply atom/browser/atom_download_manager_delegate.cc]: I mean `const content::DownloadTargetCallback& callback` here

File: atom/browser/atom_download_manager_delegate.cc [hunk 24]
```diff
@@ -333,13 +343,19 @@ void AtomDownloadManagerDelegate::OnDownloadTargetDetermined(
   }
 
   file_type_info.include_all_files = true;
+  DownloadTargetInfo* result(new DownloadTargetInfo);
```
Comment (darkdh) [inline atom/browser/atom_download_manager_delegate.cc]: this will cause memory leak.  try `base::Passed(std::move(target_info))`

**Comment by darkdh** [comment]
can you rebase to solve the conflict?

File: atom/browser/BUILD.gn [hunk 25]
```diff
@@ -22,6 +22,8 @@ source_set("browser") {
   ]
 
   deps = [
+    "//electron/chromium_src/chrome/browser/prerender",
```
Comment (bridiver) [inline atom/browser/BUILD.gn]: I don't think these should be necessary if chromium_src:safe_browser (mentioned below) is a public dep of chromium_src:browser

File: chromium_src/BUILD.gn [hunk 26]
```diff
@@ -147,12 +147,13 @@ source_set("browser") {
     ":password_manager",
     ":sessions",
     ":web_data",
-    "chrome/browser/safe_browsing",
     "//electron/brave/browser:tab_manager",
     "//base",
+    "//chrome/browser/safe_browsing",
```
Comment (bridiver) [inline chromium_src/BUILD.gn]: this should go in a new `safe_browsing` target

File: chromium_src/BUILD.gn [hunk 27]
```diff
@@ -305,8 +321,18 @@ source_set("browser") {
...
+    "//chrome/browser/interstitials/chrome_metrics_helper.h",
+
+    "//chrome/browser/renderer_preferences_util.cc",
+    "//chrome/browser/renderer_preferences_util.h",
+
+    "//chrome/browser/ssl/captive_portal_metrics_recorder.cc",
+    "//chrome/browser/ssl/captive_portal_metrics_recorder.h",
     "//chrome/browser/ssl/security_state_tab_helper.cc",
     "//chrome/browser/ssl/security_state_tab_helper.h",
+    "//chrome/browser/sync/user_event_service_factory.cc",
```
Comment (bridiver) [inline chromium_src/BUILD.gn]: these should go in the `sync` target

File: chromium_src/BUILD.gn [hunk 28]
```diff
@@ -197,12 +197,28 @@ source_set("browser") {
     "//chrome/browser/chrome_device_client.h",
 
     # extensions
+    "//chrome/browser/extensions/api/experience_sampling_private/experience_sampling.cc",
+    "//chrome/browser/extensions/api/experience_sampling_private/experience_sampling.h",
+    "//chrome/browser/extensions/extension_management.cc",
```
Comment (bridiver) [inline chromium_src/BUILD.gn]: why are these needed?

File: chromium_src/chrome/browser/ui/browser.h [hunk 29]
```diff
@@ -60,6 +60,7 @@ class Browser : public content::WebContentsDelegate {
   struct CreateParams {
     explicit CreateParams(Profile* profile);
     CreateParams(Type type, Profile* profile);
+    CreateParams(Type type, Profile* profile, bool user_gesture) {}
```
Comment (bridiver) [inline chromium_src/chrome/browser/ui/browser.h]: why is this here? We definitely don't want to implement it as a no-op either way

File: patches/master_patch.patch [hunk 30]
```diff
@@ -512,6 +566,50 @@ index d85ddf9db90d332a31f839e7600b8d6e5ec61356..2dbe745b19001ce55631e625b6c597b2
...
    int FindTabStripModelById(int32_t tab_id, TabStripModel** model) const;
+diff --git a/chrome/browser/safe_browsing/chrome_password_protection_service.cc b/chrome/browser/safe_browsing/chrome_password_protection_service.cc
+index 4275342f9244aa005521882c23767662fa2a681c..de4b9e02182dd4ed34e248d993c2a5f1e480d020 100644
+--- a/chrome/browser/safe_browsing/chrome_password_protection_service.cc
++++ b/chrome/browser/safe_browsing/chrome_password_protection_service.cc
+@@ -672,8 +672,7 @@ bool ChromePasswordProtectionService::UserClickedThroughSBInterstitial(
+ }
+ 
+ AccountInfo ChromePasswordProtectionService::GetAccountInfo() {
+-  SigninManagerBase* signin_manager =
```
Comment (bridiver) [inline patches/master_patch.patch]: stub `SigninManagerFactory::GetForProfileIfExists` instead of patching here. See site_engagement_service.cc for an example. You don't have to stub any other `SigninManagerBase` method because they will be stripped

File: patches/master_patch.patch [hunk 31]
```diff
@@ -645,7 +729,7 @@ diff --git a/chrome/common/chrome_paths_mac.mm b/chrome/common/chrome_paths_mac.
 index cca3bbb41ae25484c3e87c9aba09a8b07741e2b2..57abbcfd90dab7bbf1b5dffc35b17d3ade1a5e1e 100644
 --- a/chrome/common/chrome_paths_mac.mm
```
Comment (bridiver) [inline patches/master_patch.patch]: please remove these extraneous changes to the patches that are not related to your changes

File: chromium_src/BUILD.gn [hunk 32]
```diff
@@ -766,6 +800,8 @@ source_set("sync") {
   configs += [ ":chromium_src_config" ]
 
   sources = [
+    "//chrome/browser/sync/user_event_service_factory.cc",
```
Comment (bridiver) [inline chromium_src/BUILD.gn]: these should also go in `sync` and add a dep in downloads for them

File: brave/browser/brave_content_browser_client.cc [hunk 33]
```diff
@@ -658,7 +658,9 @@ void BraveContentBrowserClient::OverrideWebkitPrefs(
   prefs->hyperlink_auditing_enabled = false;
   // Custom preferences of guest page.
   auto web_contents = content::WebContents::FromRenderViewHost(host);
-  atom::WebContentsPreferences::OverrideWebkitPrefs(web_contents, prefs);
+  if (web_contents) {
```
Comment (bridiver) [inline brave/browser/brave_content_browser_client.cc]: when is this null?

File: brave/browser/password_manager/brave_password_manager_client.cc [hunk 34]
```diff
@@ -395,8 +395,11 @@ void BravePasswordManagerClient::CheckSafeBrowsingReputation(
     const GURL& form_action,
     const GURL& frame_url) {}
 
+void BravePasswordManagerClient::LogPasswordReuseDetectedEvent() {}
```
Comment (bridiver) [inline brave/browser/password_manager/brave_password_manager_client.cc]: ? cc @darkdh

File: atom/browser/api/atom_api_download_item.cc [hunk 35]
```diff
@@ -10,13 +10,14 @@
...
 #include "atom/common/native_mate_converters/gurl_converter.h"
+#include "atom/common/node_includes.h"
 #include "base/strings/utf_string_conversions.h"
 #include "base/threading/thread_task_runner_handle.h"
+#include "content/browser/download/download_item_impl.h"
+#include "content/public/browser/download_item.h"
 #include "native_mate/dictionary.h"
 #include "net/base/filename_util.h"
 
-#include "atom/common/node_includes.h"
```
Comment (bridiver) [inline atom/browser/api/atom_api_download_item.cc]: this is intentionally added at the end of the list

File: atom/browser/extensions/atom_extensions_browser_client.cc [hunk 36]
```diff
@@ -67,12 +67,10 @@
 #include "base/strings/string_number_conversions.h"
 #include "base/strings/stringprintf.h"
 #include "chrome/browser/extensions/api/cryptotoken_private/cryptotoken_private_api.h"
-#include "extensions/common/file_util.h"
```
[the remark this answers is not in the document]
Comment (bridiver) [reply atom/browser/extensions/atom_extensions_browser_client.cc]: it's best not to mix in unrelated changes because it just confuses the purpose of the PR. Unrelated changes should be put in a separate PR

File: atom/browser/atom_download_manager_delegate.h [hunk 37]
```diff
@@ -39,25 +50,65 @@ class AtomDownloadManagerDelegate : public content::DownloadManagerDelegate {
...
+                        const base::FilePath& suggested_virtual_path,
+                        const NotifyExtensionsCallback& callback) override {}
+  void ReserveVirtualPath(
+      content::DownloadItem* download,
+      const base::FilePath& virtual_path,
+      bool create_directory,
+      DownloadPathReservationTracker::FilenameConflictAction conflict_action,
+      const ReservedPathCallback& callback) override;
+
+  void RequestConfirmation(content::DownloadItem* download,
```
Comment (bridiver) [inline atom/browser/atom_download_manager_delegate.h]: you can't override these and ignore the callback. That's likely why you need all the patches on `download_target_determiner.cc`

File: chromium_src/BUILD.gn [hunk 38]
```diff
@@ -616,6 +623,13 @@ source_set("browser") {
   }
 }
 
+source_set("permissions") {
```
Comment (bridiver) [inline chromium_src/BUILD.gn]: all source sets need the electron_config and chromium_src_config or very subtle errors can be introduced

File: patches/master_patch.patch [hunk 39]
```diff
@@ -1788,6 +1825,19 @@ index 2ef77f4f6359618be7b37d0804a77dd4883dbc06..47f9059b1819810fc4284a9d63aae1d8
...
    proto.attachedCallback = function() {
+diff --git a/google_apis/google_api_keys.cc b/google_apis/google_api_keys.cc
+index 2e8437ed136e682540835ef4738e3519f9cb6051..e9fae5e344bbdeaa6a8348db3ec088f06b3967f6 100644
+--- a/google_apis/google_api_keys.cc
++++ b/google_apis/google_api_keys.cc
+@@ -31,7 +31,7 @@
+ #define DUMMY_API_TOKEN "dummytoken"
+ 
+ #if !defined(GOOGLE_API_KEY)
+-#define GOOGLE_API_KEY DUMMY_API_TOKEN
```
Comment (bridiver) [inline patches/master_patch.patch]: we don't want to hard code keys like this, what is the purpose of this change?

File: chromium_src/chrome/browser/safe_browsing/BUILD.gn [hunk 40]
```diff
@@ -1,19 +1,44 @@
+source_set("install_verification") {
+  sources = [
```
Comment (bridiver) [inline chromium_src/chrome/browser/safe_browsing/BUILD.gn]: same thing with source set configs here

File: patches/master_patch.patch [hunk 41]
```diff
@@ -162,6 +162,26 @@ index e2faffbfcb27da0329d3e388fea8758d49139e8e..d8cdceffbc4b048395957dcae935a4c4
...
+   DCHECK(!virtual_path_.empty());
++
+   next_state_ = STATE_CHECK_VISITED_REFERRER_BEFORE;
+ 
+   // If user has validated a dangerous download, don't check.
+@@ -730,6 +731,7 @@ void DownloadTargetDeterminer::CheckVisitedReferrerBeforeDone(
+   danger_level_ = GetDangerLevel(
+       visited_referrer_before ? VISITED_REFERRER : NO_VISITS_TO_REFERRER);
+   if (danger_level_ != DownloadFileType::NOT_DANGEROUS &&
++      danger_level_ != DownloadFileType::ALLOW_ON_USER_GESTURE &&
```
Comment (bridiver) [inline patches/master_patch.patch]: can you explain this change? We can set the user gesture just like chrome does

File: patches/master_patch.patch [hunk 42]
```diff
@@ -512,6 +532,23 @@ index d85ddf9db90d332a31f839e7600b8d6e5ec61356..2dbe745b19001ce55631e625b6c597b2
...
    int FindTabStripModelById(int32_t tab_id, TabStripModel** model) const;
+diff --git a/chrome/browser/safe_browsing/download_protection/check_client_download_request.cc b/chrome/browser/safe_browsing/download_protection/check_client_download_request.cc
+index 7d32a4cc7a0f1d2230d1f965431a291424ae0db1..ed699539f22fad86ad581553974079c9b68038a9 100644
+--- a/chrome/browser/safe_browsing/download_protection/check_client_download_request.cc
++++ b/chrome/browser/safe_browsing/download_protection/check_client_download_request.cc
+@@ -293,10 +293,12 @@ bool CheckClientDownloadRequest::IsSupportedDownload(
+     return false;
+   }
+   // This check should be last, so we know the earlier checks passed.
++#if defined(MUON_CHROMIUM_BUILD)
```
Comment (bridiver) [inline patches/master_patch.patch]: also why this change?

File: muon/browser/muon_browser_process_impl.cc [hunk 43]
```diff
@@ -11,6 +11,7 @@
 #include "chrome/browser/io_thread.h"
 #include "chrome/browser/profiles/profile.h"
 #include "chrome/browser/profiles/profile_manager.h"
+#include "chrome/browser/safe_browsing/safe_browsing_service.h"
```
Comment (bridiver) [inline muon/browser/muon_browser_process_impl.cc]: ?

File: chromium_src/chrome/browser/ui/browser.cc [hunk 44]
```diff
@@ -52,6 +52,10 @@ Browser::CreateParams::CreateParams(Type type, Profile* profile)
 
 Browser::CreateParams::CreateParams(const CreateParams& other) = default;
 
+content::WebContents* Browser::OpenURL(const content::OpenURLParams& params) {
+  return NULL;
```
Comment (bridiver) [inline chromium_src/chrome/browser/ui/browser.cc]: I'm a little worried about this one causing a crash. Where does it get called from? Is it just windows doing its usual terrible job of dead code elimination or is there an actual legit code path that leads here?

File: atom/browser/atom_resource_dispatcher_host_delegate.cc [hunk 45]
```diff
@@ -79,12 +87,66 @@ void HandleExternalProtocolInUI(
...
+    net::HttpRequestHeaders headers;
+    headers.CopyFrom(request->extra_request_headers());
+    request->SetExtraRequestHeaders(headers);
+  }
+
+  AppendStandardResourceThrottles(request,
+                                  resource_context,
+                                  resource_type,
+                                  throttles);
+#if BUILDFLAG(ENABLE_NACL)
```
Comment (bridiver) [inline atom/browser/atom_resource_dispatcher_host_delegate.cc]: I don't think this should be here
